internal/attacktechniques/azure/execution/vm-run-command/main.tf

```diff
@@ -0,0 +1,117 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.8.0"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# Random
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+resource "random_string" "lab_name" {
+  length  = 8
+  special = false
+  upper   = false
+}
+
+resource "random_password" "password" {
+  length           = 64
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# Resource Group
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+resource "azurerm_resource_group" "lab_environment" {
+  name     = "rg-${random_string.lab_name.result}"
+  location = "West US"
+}
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# Networking Resources
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+resource "azurerm_virtual_network" "lab_vnet" {
+  name                = "vnet-${random_string.lab_name.result}"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.lab_environment.location
+  resource_group_name = azurerm_resource_group.lab_environment.name
+}
+
+resource "azurerm_subnet" "lab_subnet" {
+  name                 = "subnet-${random_string.lab_name.result}"
+  resource_group_name  = azurerm_resource_group.lab_environment.name
+  virtual_network_name = azurerm_virtual_network.lab_vnet.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_network_interface" "lab_nic" {
+  name                = "nic-${random_string.lab_name.result}"
+  location            = azurerm_resource_group.lab_environment.location
+  resource_group_name = azurerm_resource_group.lab_environment.name
+
+  ip_configuration {
+    name                          = "ip-${random_string.lab_name.result}"
+    subnet_id                     = azurerm_subnet.lab_subnet.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# Virtual Machine Resources
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+resource "azurerm_windows_virtual_machine" "lab_windows_vm" {
+  name                = "vm-${random_string.lab_name.result}"
+  resource_group_name = azurerm_resource_group.lab_environment.name
+  location            = azurerm_resource_group.lab_environment.location
+  size                = "Standard_F2"
+  admin_username      = "local_admin_user"
+  admin_password      = random_password.password.result
+  user_data           = base64encode(random_string.lab_name.result)
```

Comment:
Not sure what's the use for this?

Reply:
Ah! This is a left-over from the original terraform that I modified for this PR. But it is worth mentioning why I originally needed it because we will likely need to do the same in the future.

In Azure, key vaults are addressable by domain `https://<my_vault_name>.vault.azure.net`. If we create a virtual machine and assign it a managed identity with the rights to read secret from the key vault, we need to know the name of the vault so we can get the secrets from it. By injecting the name of the (randomly generated) key vault into user data, we can retrieve it from the IDMS on the virtual machine. For example:

```powershell
# Fetch the randomly generated lab ID from the UserData in IMDS
# We need this to assemble the key vault URI, which we will hit
# to request the secrets from the vault.

Write-Host "    [+] Fetching Secrets from the Keyvault"
$userData = Invoke-RestMethod -Method GET -Headers @{"Metadata" = "true" } -Uri "http://169.254.169.254/metadata/instance/compute/userData?api-version=2021-01-01&format=text"
$labId = [System.Text.Encoding]::UTF8.GetString([Convert]::FromBase64String($userData))

# Next we get an actual access token using the managed identity
# with the key vault as the audience.

$imdsResponse = Invoke-RestMethod -Uri 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fvault.azure.net' -Method GET -Headers @{Metadata = "true" }
$keyVaultToken = $imdsResponse.access_token

# Fetch the secrets from the key vault. Each secret is returned
# in the `value` property of the response.

$keyVaultUrl = "https://vault-$labId.vault.azure.net"
$mySecret = Invoke-RestMethod -Uri "$keyVaultUrl/secrets/my-secret?api-version=2016-10-01" -Method GET -Headers @{Authorization = "Bearer $keyVaultToken" }

```

There may be a more elegant way of doing through by enumerating vaults via API, but this seemed simpler to me.

We can remove that line, but I figured I would share the reasoning.